src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -252,6 +252,72 @@ void bulk_load_service::create_partition_bulk_load_dir(const std::string &app_na
 
 // ThreadPool: THREAD_POOL_META_STATE
 void bulk_load_service::partition_bulk_load(const std::string &app_name, const gpid &pid)
+{
+    FAIL_POINT_INJECT_F("meta_bulk_load_partition_bulk_load", [](dsn::string_view) {});
+
+    rpc_address primary_addr;
+    ballot b;
+    {
+        zauto_read_lock l(app_lock());
+        std::shared_ptr<app_state> app = _state->get_app(pid.get_app_id());
+        if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+            dwarn_f("app(name={}, id={}) is not existed, set bulk load failed",
+                    app_name,
+                    pid.get_app_id());
+            handle_app_unavailable(pid.get_app_id(), app_name);
+            return;
+        }
+        primary_addr = app->partitions[pid.get_partition_index()].primary;
+        b = app->partitions[pid.get_partition_index()].ballot;
+    }
+
+    if (primary_addr.is_invalid()) {
+        dwarn_f("app({}) partition({}) primary is invalid, try it later", app_name, pid);
```

Comment:
If the primary_addr is invalid for a very long time, does bulk_load_service::partition_bulk_load will retry forever?

Reply:
`primary_addr` may be invalid temporarily during replica primary switch, it will not be invalid for a very long time.